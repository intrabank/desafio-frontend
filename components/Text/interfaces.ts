import { ReactNode } from "react"

export interface TitleProps {
    children: ReactNode
}

export interface MediumTitleProps extends TitleProps {
}

export interface ParagraphProps extends TitleProps {
    size?: number
}