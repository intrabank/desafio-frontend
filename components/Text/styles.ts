import { styled } from "@stitches/react";

const white = '#fff'

export const StyledTitle = styled('h1', {
    color: white,
    fontSize: '45px',
    fontWeight: '700'

})

export const StyledMediumTitle = styled('h1', {
    color: white,
    fontSize: '25px',
    fontWeight: '600'

})

export const StyledParagraph = styled('p', {
    color: white,
    display: 'block',
    fontSize: '$customSize',
    fontWeight: '300'
})

export const StyledPhrase = styled('span', {
    fontSize: '$customSize',
    display: 'block'
})

export const StyledError = styled('p', {
    fontSize: '12px',
    color: '#FA4D56',
})