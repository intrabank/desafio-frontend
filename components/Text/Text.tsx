import { StyledError, StyledMediumTitle, StyledParagraph, StyledPhrase, StyledTitle } from "./styles";
import { MediumTitleProps, ParagraphProps, TitleProps } from "./interfaces";

const TextComponent = ({ text }: any) => {
  return <p>{text}</p>;
};

const Title = ({ children }: TitleProps) => {
  return <StyledTitle>{children}</StyledTitle>;
};

const MediumTitle = ({ children }: MediumTitleProps) => {
  return <StyledMediumTitle>{children}</StyledMediumTitle>;
};

const Paragraph = ({ children, size }: ParagraphProps) => {
  return <StyledParagraph css={{fontSize: `${size}px`}}>{children}</StyledParagraph>;
};

const Phrase = ({ children, size }: ParagraphProps) => {
  return <StyledPhrase css={{fontSize: `${size}px`}}>{children}</StyledPhrase>;
};

const ErrorTxt = ({children}: ParagraphProps) => {
  return <StyledError>{children}</StyledError>
}

TextComponent.title = Title;
TextComponent.mediumTitle = MediumTitle;
TextComponent.paragraph = Paragraph;
TextComponent.phrase = Phrase;
TextComponent.error = ErrorTxt;

export default TextComponent;
