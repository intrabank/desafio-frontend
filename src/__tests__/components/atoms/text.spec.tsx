import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Text from '../../../components/atoms/text';
import renderer from 'react-test-renderer';

describe('Text', () => {
	it('should render text', () => {
		const tree = renderer.create(<Text>Test</Text>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should change with variants', () => {
		render(
			<Text as='h1' size='xl' weight='bold'>
				Test
			</Text>
		);
		const text = screen.getByText('Test');
		const styles = getComputedStyle(text);

		expect(styles.fontSize).toBe('2em');
		expect(styles.fontWeight).toBe('bold');
		expect(text.tagName).toBe('H1');
	});
});
