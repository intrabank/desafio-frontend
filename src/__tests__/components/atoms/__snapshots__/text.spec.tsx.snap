// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render text 1`] = `
<p
  className="c-eFhfvJ c-eFhfvJ-fPSeYF-color-white c-eFhfvJ-gJpNlm-size-md c-eFhfvJ-hZNfDR-weight-medium"
>
  Test
</p>
`;
